container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/user_services_functions/start_user_services.go

```diff
@@ -169,6 +173,27 @@ func StartRegisteredUserServices(
 		return nil, nil, stacktrace.Propagate(err, "Couldn't get an object attribute provider for enclave '%v'", enclaveUuid)
 	}
 
+	// Check if the logs collector is available
+	// As the container logs are sent asynchronously we'd not know whether they're being received by the collector and there would be no errors if the collector never comes up
+	// The least we can do is check if the collector server is healthy before starting the user service, if in case it gets shut down later we can't do much about it anyway.
+	if err = logsCollectorAvailabilityChecker.WaitForAvailability(); err != nil {
+		return nil, nil,
+			stacktrace.Propagate(err, "An error occurred while waiting for the log container to become available")
+	}
+
+	//We use the public TCP address because the logging driver connection link is from the Docker demon to the logs collector container
+	//so the direction is from the host machine to the container inside the Docker cluster
```

Comment:
What does it mean "public TCP address" here? TCP is just a protocol, and what does "public" mean?